### PR TITLE
oh-my-git: init at 0.6.4

### LINKED ISSUE #112341: Please package oh-my-git
**Project description**

>  An interactive Git learning game!

made with godot engine

it does not run with steam-run (https://github.com/git-learning-game/oh-my-git/issues/2), so a proper package would be great

**Metadata**

* homepage URL: https://blinry.itch.io/oh-my-git
* source URL: https://github.com/git-learning-game/oh-my-git
* license: Blue Oak Model License
* platforms: linux, darwin, windows

### PR DESCRIPTION
**Pull request by jojosch**
###### Motivation for this change
Add the package for @davidak's package request. Closes #112341. I played about half of the levels and there seem to be no problems running the game.

The godot export commands expects a copy of the godot export templates at `$HOME/.local/share/godot/templates/3.2.3.stable/...`. First i downloaded the zip archive from https://godotengine.org/download and copied the templates to the expected location. But the version of these export templates and the godot compiler need to be in sync.

Therefore i added `godot-export-templates` based on the `headless`/`server` package. Now the version of godot and the export templates are automatically in sync.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
```
/nix/store/xx9zihn59643p47b2qgyl05qxk4b8qrq-oh-my-git-0.6.4	 396.7M
/nix/store/hk6l3krh3aps7a6pkxbf6p1kp1zyvbc4-godot-export-templates-3.2.3	 391.6M
```
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119642` at cc6cd3c8 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>godot-export-templates</li>
    <li>oh-my-git</li>
  </ul>
</details>

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 119642` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>godot-export-templates</li>
    <li>oh-my-git</li>
  </ul>
</details>

File: pkgs/development/tools/godot/export-templates.nix
Comment (davidak) [inline pkgs/development/tools/godot/export-templates.nix:11]: does x11 mean it will not work on wayland?

File: pkgs/games/oh-my-git/default.nix
Comment (davidak) [inline pkgs/games/oh-my-git/default.nix:110]: that's a good start. can this package also work on arm, riscv, darwin, windows with few changes?

**Review by davidak — CHANGES_REQUESTED** [review]
Thanks for creating the packages! :sparkles: 

You should add git as runtime dependency. Someone who plays this game to learn git might not have it installed already.

tested with:

```
[nix-shell:~/.cache/nixpkgs-review/pr-119642]$ nix-shell -p --pure

[nix-shell:~/.cache/nixpkgs-review/pr-119642]$ git
The program 'git' is not in your PATH. It is provided by several packages.
You can make it available in an ephemeral shell by typing one of the following:
  nix-shell -p git
  nix-shell -p git-doc
  nix-shell -p gitMinimal
  nix-shell -p gitSVN

[nix-shell:~/.cache/nixpkgs-review/pr-119642]$ results/oh-my-git/bin/oh-my-git
```

Screenshot from 2021-04-18 05-17-58

Change looks otherwise good.

Builds and works. Even finds my previous savegame :)

File: pkgs/development/tools/godot/export-templates.nix
Comment (jojosch) [reply pkgs/development/tools/godot/export-templates.nix:11]: I'm running sway as my "daily driver" - there it works fine (running with XWayland, confirmed that with `xeyes`).

File: pkgs/games/oh-my-git/default.nix
Comment (jojosch) [reply pkgs/games/oh-my-git/default.nix:112]: I can try to get it to run under darwin. For that i need to adjust `godot*` (setting the build `platform`, conditionally adding the linux/darwin specific dependencies (e.g `alsaLib`) and so on.

Comment (davidak) [reply pkgs/games/oh-my-git/default.nix:112]: it would be fine with me to add it in a later pr, so we can get this merged asap

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 119642` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>godot-export-templates</li>
    <li>oh-my-git</li>
  </ul>
</details>

**Review by davidak — APPROVED** [review]
Looks good

Builds, works

**Comment by davidak** [comment]
@SuperSandro2000 do you want to add a second review and merge?

File: pkgs/games/oh-my-git/default.nix
Comment (jojosch) [reply pkgs/games/oh-my-git/default.nix:112]: Yeah, i tried to build `godot` under darwin a few minutes ago but it failes to find the header files for e.g. `AudioUnit` and so on. I added them the same way as e.g. in

https://github.com/NixOS/nixpkgs/blob/831e64bad4616a5be3281537c431087be2d372f8/pkgs/applications/video/jellyfin-media-player/default.nix#L61-L66

Will need to investigate that further.

**Pull request merged**
